### PR TITLE
CSHARP-1906: Support for case insensitive search using $in clause in Linq expressions

### PR DESCRIPTION
**Pull request by mpbelov**
https://jira.mongodb.org/browse/CSHARP-1906

This pull request adds the support of case insensitive search using $in clause in Linq expressions like it works for general equal clause. For example it might be done in this way:

var goodValues = new string[] { ... }
var query = collection.AsQueryable<T>()
    .Where(i => goodValues.Contains(i.SomeField.ToLower()));

Right now it's only possible to specify general values in the array and is not possible to specify regex values for case insensitive search. General equal clause suports it via ToLower() etc methods, but the driver throw an exception if there is a call of .ToLower() method inside the array.Contains() method in .Where() expression.

### CODE REVIEW

**Comment by mpbelov** [comment]
Does anybody in MongoDB community check incoming pull requests? It's been half a year: no comments, no review, nothing.

**Comment by mpbelov** [comment]
@craiggwilson @rstam could you please look at the pull request?

**Comment by jsheely** [comment]
Bueler....... Bueler.....

This would be nice to have merged in.

**Comment by scmfabrin** [comment]
Hello, 
are there some news about that request?

**Comment by YNathan** [comment]
hi there do you have any solution for this problem? unbelievable I try it with 8 ways and didn't work

**Comment by thefat32** [comment]
In 2021 this is still a problem, what is need to be done to get this merged? A lot of people in forums is asking about this feature. I would be glad to help fixing any code or conflict.

**Comment by JamesKovacs** [comment]
Please see [this JIRA comment](https://jira.mongodb.org/browse/CSHARP-1906?focusedCommentId=4043830&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4043830) for an update on the current status.

TL;DR - A new LINQ provider is nearing completion which fixes this problem and many others.
